### PR TITLE
build: upgrade to husky 7

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [x] Other... Please describe: CI

## What is the current behavior?

admins can always push to main branch

## What is the new behavior?

```
$ git checkout main
$ git commit -m "push to master" 
-e 
🚫 Cannot push to remote main branch, please create your own branch and use PR.
husky - pre-commit hook exited with code 1 (error)
```

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+git update-index --again
+
+# shellcheck disable=SC2006
+BRANCH=`git rev-parse --abbrev-ref HEAD`
+PROTECTED_BRANCHES="^(main|gh-pages)"
+
+# shellcheck disable=SC2039
+if [[ "$BRANCH" =~ $PROTECTED_BRANCHES ]]
+then
+  echo -e "\n🚫 Cannot push to remote $BRANCH branch, please create your own branch and use PR." && exit 1
```
Comment (IKatsuba) [inline .husky/pre-commit:14]: а зачем? И так ничего не запушится в протектед бранчи, а тут список нужно актуализировать

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+fi
+
+npm run lint-staged
```
Comment (IKatsuba) [inline .husky/pre-commit:17]: кажется проще `npx` заюзать, что бы не прокидывать такие вещи в npm scripts

Comment (splincode) [reply .husky/pre-commit:17]: npx is deprecated from npm@6, in future npm@7 you can use `npm exec --`

File: .husky/pre-commit (hunk 1 above)
Comment (splincode) [reply .husky/pre-commit:14]: Sorry, but you can (( 

I was published my commits to main branch by chance

image

File: .husky/pre-commit (hunk 2 above)
Comment (IKatsuba) [reply .husky/pre-commit:17]: npx deprecated как standalone решение. Но оно все еще будет работать и под капотом вызывать npm exec

Comment (splincode) [reply .husky/pre-commit:17]: No problem, I will fix it))

Comment (splincode) [reply .husky/pre-commit]: @IKatsuba fixed

File: .husky/pre-commit (hunk 1 above)
Comment (waterplea) [reply .husky/pre-commit:14]: @IKatsuba all admins can push to main

Comment (IKatsuba) [reply .husky/pre-commit:14]: think there are too many admins

Comment (splincode) [reply .husky/pre-commit:14]: this does not exclude the fact that you can commit and push to main branch, although you did not want to ))

Comment (splincode) [reply .husky/pre-commit:14]: this script will simply prohibit doing this, at the terminal or IDEA

**Comment by splincode** [comment]
@waterplea @IKatsuba updated :)

**Review by IKatsuba — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**